Firebase/Storage/FIRStorageUploadTask.m

```diff
@@ -177,6 +181,37 @@ - (void)enqueue {
   }];
 }
 
+- (void)finishTaskWithStatus:(FIRStorageTaskStatus)status
+                    snapshot:(FIRStorageTaskSnapshot *)snapshot {
+  [self fireHandlersForStatus:status snapshot:self.snapshot];
+  [self removeAllObservers];
+  self->_fetcherCompletion = nil;
+}
+
+- (BOOL)isContentToUploadValid:(NSError **)outError {
+  if (_uploadData != nil) {
+    return YES;
+  }
+
+  NSError *fileReachabilityError;
+  if (![_fileURL checkResourceIsReachableAndReturnError:&fileReachabilityError]) {
+    if (outError != NULL) {
+      NSString *description =
+          [NSString stringWithFormat:@"File at URL: %@ is not reachable.", _fileURL.absoluteString];
+      *outError = [NSError errorWithDomain:FIRStorageErrorDomain
+                                      code:FIRStorageErrorCodeUnknown
+                                  userInfo:@{
+                                    NSUnderlyingErrorKey : fileReachabilityError,
```

Comment:
I'm trying to upload an mp4 file from Flutter with `firebase_storage` (got the file from `image_picker`, seems to work fine with jpeg from `image_picker`) and I'm hitting an app crash here.

`fileReachabilityError` is `nil` and thus I'm getting an `NSInvalidException` since you cannot add `nil` values to an `NSDitctionary`.

For some reason `_fileURL` is `nil`, which makes `[_fileURL checkResourceIsReachableAndReturnError:&fileReachabilityError]` return just `nil`, which triggers the `if (!...)` branch.

Sorry for not providing too much info, I'm still in the process of tracking it down but thought it would be good to see if anyone had any ideas ☺️ 

ping @maksymmalyhin

Reply:
@LinusU Thank you for the report. That's a good case. Yeah, this part of code definitely does not expect `_fileURL` to be `nil`. We will fix it.

In the meanwhile, would you be able to post a comment about it to #2350 or create a separate ticket, so it is not lost.

As a workaround you should check the `fileURL != nil` before passing it to the `putFile` method.

Reply:
I've submitted flutter/plugins#1504 and flutter/plugins#1505 to fix this in Flutter, but would be great to see a fix here as well ☺️ 

I'll put a comment in the issue you linked!

Reply:
@LinusU thanks!